hydrolib/core/io/dflowfm/structure/models.py

```diff
@@ -108,27 +108,62 @@ def check_location(cls, values: dict) -> dict:
         filtered_values = {k: v for k, v in values.items() if v is not None}
         structype = filtered_values.get("type", "").lower()
```

Comment:
Do you think it is possible to migrate the structure location validation to the "new" general location validation: `get_location_specification_rootvalidator`?

We can add pli to that easily.

Reply:
If it's fine with you, shall we continue this under #436?

Reply:
Sure!